mwapi/session.py

```diff
@@ -60,15 +69,33 @@ def __init__(self, host, user_agent=None, api_path="/w/api.php",
             self.headers['User-Agent'] = user_agent
 
     def _request(self, method, params=None, data=None, files=None):
-        resp = self.session.request(method, self.api_url, params=params,
-                                    data=data, files=files,
-                                    headers=self.headers, stream=True)
+        try:
+            resp = self.session.request(method, self.api_url, params=params,
+                                        data=data, files=files,
+                                        timeout=self.timeout, headers=self.headers,
+                                        stream=True)
+        except requests.exceptions.Timeout as e:
+            raise TimeoutError(str(e))
+        except requests.exceptions.ConnectionError as e:
+            raise ConnectionError(str(e))
+        except requests.exceptions.HTTPError as e:
+            raise HTTPError(str(e))
+        except requests.exceptions.TooManyRedirects as e:
+            raise TooManyRedirectsError(str(e))
+        except requests.exceptions.RequestError as e:
+            raise RequestError(str(e))
+        except Exception as e:
+            raise RequestError(str(e))
```

Comment:
Should definitely not stringify the (e) - that'll lose information in the underlying exception that'l be of help...